tests/integration-tests/tests/storage/test_deletion_policy.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+
+import boto3
+import pytest
+from assertpy import assert_that
+from botocore.exceptions import ClientError
+from utils import get_compute_nodes_instance_ids, get_root_volume_id, get_stack_id_tag_filter
+
+from tests.tags.test_tag_propagation import get_head_node_instance_id
+
+
+@pytest.mark.usefixtures("instance", "scheduler")
+def test_retain_on_deletion(pcluster_config_reader, clusters_factory, region, os):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+
+    stack_arn = cluster.cfn_stack_arn
+    retained_volume = cluster.cfn_resources["EBS0"]
+    head_node_instance_id = get_head_node_instance_id(cluster)
+    head_node_root_volume = get_root_volume_id(head_node_instance_id, region, os)
+    compute_node_instance_ids = get_compute_nodes_instance_ids(cluster.name, region)
+    logging.info("Checking at least one compute node is running")
+    assert_that(len(compute_node_instance_ids)).is_greater_than_or_equal_to(1)
+    compute_root_volumes = []
+    for compute_node in compute_node_instance_ids:
+        compute_root_volumes.append(get_root_volume_id(compute_node, region, os))
+    logging.info("Compute root volume %s", compute_root_volumes)
+
+    ec2_client = boto3.client("ec2")
```

Comment:
region_name?

Reply:
Added

Reply:
In individual test, it is very safe to not add the region_name, because it is set in the environment in through a fixture

Reply:
you are right I had missed we have that fixture